.github/workflows/ci.yml

```
@@ -8,6 +8,11 @@ on:
       - 'maintainers/**'
   workflow_dispatch:
 
+# Checks if any concurrent jobs under the same pull request or branch are being executed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

This will cause CI runs on the master branch to be cancelled if multiple PRs are merged in quick succession. I believe it'd be good to have all jobs on master finish, since it could make it easier to track down an error. Maybe we can slightly tweak the group to something like this:
```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

`github.head_ref` is only available on PRs, which means that all other sources will use `github.run_id`, making them unique. WDYT?

---

But then CI running on a branch (ie: not pr) won't be part of the concurrency group, right?

---

It won't but the only branches we have this workflow set to run on are `master`, `release/` and `maintainers/`, I wouldn't think those are branches with high volume of pushes

---

Just to clarify, pushing to a branch on a PR triggers as a PR, not as a push.

---

Agree, it is the best we can do.
Thank you for your valuable inputs Mauro!